src/tools/ddsperf/ddsperf.c

```diff
@@ -1427,11 +1436,25 @@ static bool print_stats (dds_time_t tref, dds_time_t tnow, dds_time_t tprev, str
 
     if (nrecv > 0 || substat_every_second)
     {
+      if (filename != NULL)
+      {
+        if (file == NULL)
+        {
+          file = fopen(filename, "w");
+        }
+      }
       const double dt = (double) (tnow - tprev);
       printf ("%s size %"PRIu32" total %"PRIu64" lost %"PRIu64" delta %"PRIu64" lost %"PRIu64" rate %.2f kS/s %.2f Mb/s (%.2f kS/s %.2f Mb/s)\n",
               prefix, last_size, tot_nrecv, tot_nlost, nrecv, nlost,
               (double) nrecv * 1e6 / dt, (double) nrecv_bytes * 8 * 1e3 / dt,
               (double) nrecv10s * 1e6 / (10 * dt), (double) nrecv10s_bytes * 8 * 1e3 / (10 * dt));
+      if (file != NULL)
+      {
+        int64_t tsec = (int64_t)(tnow / 1000000000);
+        int64_t nsec = (int64_t)(tnow % 1000000000);
+        fprintf (file, "%"PRId64".%"PRId64",%s,%"PRIdPID",%.3f,%"PRIu32",%.2f,%.2f\n",
```

Comment:
That should be `"%"PRId64".%06"PRId64"` and I suspect the `%s` should be `\"%s\"` but I don't know CSV rules very well.

I furthermore think some of the other fields that get written to `stdout` here are useful; and certainly the CPU, memory and network load data would be good to also collect in the CSV file. They are gathered and printed because they are valuable for understanding what is going on, it follows that a log file specifically for analysis purposes should include them.

Reply:
- Changed the format specifier as proposed in the comment.
- Added code to log the latency measurements to csv file.